core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -386,6 +390,15 @@ private void completeHeaders(Metadata.Headers headers) {
       headers.put(TIMEOUT_KEY, method.getTimeout());
     }
 
+    private void startDeadlineTimer() {
+      SharedResourceHolder.get(TIMER_SERVICE).schedule(new Runnable() {
```

Comment:
Each get() should be coupled with a release(), as it does resource counting. I'd suggest doing a get() during construction and doing a release() near where terminationRunnable.run() is called (it is called twice).